code/game/area/areas.dm

```diff
@@ -230,6 +230,25 @@ GLOBAL_LIST_EMPTY(teleportlocs)
 	contained_turfs -= turfs_to_uncontain
 	turfs_to_uncontain = list()
 
+/area/space/proc/cannonize_contained_turfs_slow()
+	// space has a gigantic fucking amount of turfs after loading the map
+	// Since we know it's safe to do so right now, chunk up the uncontainment
```

Comment:
Biggest gains right here. The first time /area/space called cannonize_contained_turfs() it took around 6s for the BYOND list `-=` operation

Reply:
@LemonInTheDark

Reply:
Copying off the core list seems like something that'd be pretty slow, tho I guess not cause it's not moving all that much at once?

Reply:
/area/space should not have a large amount of uncontained turfs, we put in effort when loading large spaces to add them like... in place if that makes sense.
It's strange that that isn't the case. I'd be curious to know where the problem turfs came from. ruins maybe?

Reply:
As far as speed goes, it's about the same amount of time minus the tick overruns.